storage/paths/local.go

```diff
@@ -438,7 +468,24 @@ func (st *Local) Reserve(ctx context.Context, sid storiface.SectorRef, ft storif
 
 	st.localLk.Lock()
 
-	done := func() {}
+	var firstDonebuf []byte
+	var firstDoneN int
+
+	var doneCalled bool
+	done := func() {
+		if doneCalled {
+			curStack := make([]byte, 20480)
+			curStackN := runtime.Stack(curStack, false)
+
+			log.Errorw("double done call", "sector", sid, "fileType", ft, "prevStack", string(firstDonebuf[:firstDoneN]), "curStack", string(curStack[:curStackN]))
+		}
+
+		firstDonebuf = make([]byte, 20480)
+		firstDoneN = runtime.Stack(firstDonebuf, false)
```

Comment:
firstDonebuf = firstDonebuf[:runtime.Stack(firstDonebuf, false)]
Then you can get rid of firstDoneN